include/librealsense2/hpp/rs_eth_config.hpp

```diff
@@ -0,0 +1,269 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2025 RealSense, Inc. All Rights Reserved.
+
+#ifndef LIBREALSENSE_RS2_ETH_CONFIG_HPP
+#define LIBREALSENSE_RS2_ETH_CONFIG_HPP
+
+#include "rs_types.hpp"
+#include "rs_device.hpp"
+#include "../h/rs_eth_config.h"
+
+#include <sstream>
+
+namespace rs2
+{
+    /**
+     * \brief Ethernet configuration extension for devices that support ethernet configuration
+     */
+    class eth_config_device : public device
+    {
+    public:
+        eth_config_device() : device() {}
+
+        eth_config_device( rs2::device d ) : device( d.get() )
+        {
+            rs2_error * e = nullptr;
+            // Ethernet configuration is supported through HWM commands, must be extensible to RS2_EXTENSION_DEBUG
+            if( rs2_is_device_extendable_to( _dev.get(), RS2_EXTENSION_DEBUG, &e ) == 0 && ! e )
+                _dev.reset();
+            else if( ! rs2_supports_eth_config( _dev.get(), &e ) && ! e )
```

Comment:
if() do a elseif () do a
seems this can be replaced by if ( ... OR ...)

Reply:
Style. I thought it was too complex statement. Changed to use a single `if` statement.